### PR TITLE
more robust echobot, let roboed go

### PR DESCRIPTION
**Pull request by rahiel**
I had a weird crash where I received an update without a message, so the call to `update.message.text` crashed it. This takes care of that and the case where an update leads to an `Unauthorized` exception when `len(bot.getUpdates()) > 1`.

We also say goodbye to `roboed.py`, it does not teach more than echobot and I am too lazy to update its exception handling code.

### CODE REVIEW

File: examples/echobot.py [hunk 1]
```diff
@@ -25,29 +28,29 @@ def main():
...
-def echo(bot, update_id):
-
+def echo(bot):
+    global update_id
     # Request updates after the last update_id
     for update in bot.getUpdates(offset=update_id, timeout=10):
         # chat_id is required to reply to any message
         chat_id = update.message.chat_id
         update_id = update.update_id + 1
+        if not update.message:  # we ignore updates without messages
```
Comment (leandrotoledo) [inline examples/echobot.py]: Updates without messages could be edited messages, nice catch.

File: examples/echobot.py [hunk 2]
```diff
@@ -25,29 +28,29 @@ def main():
...
+            continue
         message = update.message.text
 
         if message:
```
Comment (leandrotoledo) [inline examples/echobot.py]: Maybe for simplicity we could go with (and remove the ``if not update.message``):
```
if update.message:
   bot.sendMessage(chat_id=chat_id, text=update.message.text)
```

But it's your call.

**Comment by leandrotoledo** [comment]
Minor comments, other than that looks good to me.

``So long and thanks for all the fish roboed.py``

File: examples/echobot.py (hunk 2 above)
Comment (rahiel) [inline examples/echobot.py]: That looks much better!

**Comment by leandrotoledo** [comment]
Nailed it! Good to be merged.

**Pull request merged**